tables.go

```diff
@@ -0,0 +1,214 @@
+package fastjson
+
+import "unicode/utf8"
+
+// safeSet holds the value true if the ASCII character with the given array
+// position can be represented inside a JSON string without any further
+// escaping.
+//
+// All values are true except for the ASCII control characters (0-31), the
+// double quote ("), and the backslash character ("\").
+var safeSet = [utf8.RuneSelf]bool{
```

Comment:
I wonder if using a switch / case statement yields even better performance. The gains are good, but maybe we're leaving some on the table by not using a switch/case?

Reply:
I would expect the array to be fastest, since it's just a simple direct index operation. A switch would probably end up as a binary search.